cdap-ui/app/cdap/api/market.js

```diff
@@ -19,14 +19,21 @@ import { apiCreatorAbsPath } from 'services/resource-helper';
 
 let dataSrc = DataSourceConfigurer.getInstance();
 const basepath = window.CDAP_CONFIG.marketUrl;
+const basepaths = window.CDAP_CONFIG.marketUrls;
 // FIXME (CDAP-14836): Right now this is scattered across node and client. Need to consolidate this.
 const REQUEST_TYPE_MARKET = 'MARKET';
 const requestOptions = {
   requestOrigin: REQUEST_TYPE_MARKET,
 };
 
+function getVerifiedMarketBaseUrl(url) {
+  const found = basepaths.find((element) => element === decodeURIComponent(url));
```

Comment:
Why is the function returning empty string? This should be undefined if UI is trying fetch an asset form an unknown market right.

Reply:
nit: the input is also not url. It should be host. We are checking the host of the url to talk to right.